engine/src/main/java/org/terasology/rendering/nui/widgets/UIImageSlideshow.java

```diff
@@ -29,15 +29,16 @@
 import java.util.List;
 
 /**
- * This widget displays images in sequence in an image slideshow. Switching images automatically after a time period.
+ * This widget displays images in sequence in an image slideshow.
+ * Switching images automatically after a time period or manually.
  */
-public class UITimedImageSlideshow extends CoreWidget {
+public class UIImageSlideshow extends CoreWidget {
 
     private Binding<UIWidget> currentImage = new DefaultBinding<>();
     private List<UIImage> images = new ArrayList<>();
```

Comment:
I would like to see this have `@LayoutConfig` as well

Reply:
### Option A
Change this to a `List<TextureRegion>`

### Option B
Change this to a `List<UIWidget>` and rename appropriately

Reply:
Honestly, both these options break Single responsibility principle. I'm pretty sure that we aren't going to have another types for slideshow in future. 
So, I mean we don't need to have so generic type UIWidget for **ImageSlideshow**. It just doesn't make any sense. The same for TextureRegion. I'd try to avoid using textures in this widget, just because it shouldn't care about so specific type of content, which slideshow uses.